### ISSUE TITLE
How to access intermediate tensors that are used during the application (tf.app) lifecycle

### ISSUE THREAD

**Issue by ParikhKadam** [tensorflow/tensorflow#37133]
Basically, I am running the evaluation script, provided by Tf's Object Detection API, using:

```
python3 object_detection/legacy/eval.py \
    --pipeline_config_path=../../faster_rcnn_resnet101_voc12.pbtxt\
    --checkpoint_dir=../../trainedmodel_voc07 \
    --eval_dir=../../eval_results \
    --run_once=True \
    --alsologtostderr
```

Now, this script outputs the bounding boxes on image along with prediction scores. But what I need is, the metrics calculated.

When I looked into the eval.py, I found that, it returns nothing. It run the evaluation, calculates the metrics but doesn't return it. So, I just changed the code to return the metrics and print it on console.

```
metrics = evaluator.evaluate(
        create_input_dict_fn,
        model_fn,
        eval_config,
        categories,
        FLAGS.checkpoint_dir,
        FLAGS.eval_dir,
        graph_hook_fn=graph_rewriter_fn,
    )
print(metrics)
```

But, as we know, this won't print the values inside the metric. I have tried several things like `metrics.numpy()` which won't work when eager execution is turned off. I have also tried using `metrics.eval()` by creating a session but this just hangs up the PC and does nothing.

Also, `tf.print()` won't work as it runs in graph mode.

Then I found that TF stores summaries of metrics and we can visualize it using tensorboard. That's okay.. but what I wanted to extract values from other tensors?

So, basically, how can I access the intermediate tensors in the script?

**Comment by amahendrakar** [comment]
@ParikhKadam,
Looks like this issue is related to tensorflow/models repo. This is a duplicate of an earlier issue you have mentioned [#8209](https://github.com/tensorflow/models/issues/8209).
Can we track that issue and close this one. It will help us to follow easily. Please let us know. Thanks!

**Comment by ParikhKadam** [comment]
@amahendrakar  Maybe you should close that issue then. Because, I think this as a flaw in tf.app rather than of models repo. `tf.app` doesn't provide a way to access tensors directly (maybe).

**Comment by gowthamkpr** [comment]
@ParikhKadam Closing this issue here as it a duplicate of [#8209](https://github.com/tensorflow/models/issues/8209). Lets track it in the models repo.
